File: 02-selection-sort/python/selection_sort.py
def find_smallest_index(arr):
    smallest_index = 0
    smallest = arr[0]

    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index

def selection_sort(arr):
    new_arr = []
    for i in range(len(arr)):
        smallest_index = find_smallest_index(arr)
        new_arr.append(arr.pop(smallest_index))
    return new_arr

File: 02-selection-sort/ruby/selection_sort.rb
# frozen_string_literal: true

def find_smallest_index(arr)
  smallest_index = 0
  smallest = arr[0]

  1.upto(arr.size - 1).each do |index|
    if arr[index] < smallest
      smallest = arr[index]
      smallest_index = index
    end
  end
  smallest_index
end

def selection_sort(arr)
  new_arr = []
  arr.size.times do
    smallest_index = find_smallest_index(arr)
    new_arr << arr.delete_at(smallest_index)
  end
  new_arr
end
